cg/meta/workflow/nf_analysis.py

```diff
@@ -235,6 +300,57 @@ def write_trailblazer_config(self, case_id: str, tower_id: str) -> None:
             file_path=config_path,
         )
 
+    def create_sample_sheet(
+        self,
+        case_id: str,
+        dry_run: bool,
+    ):
+        """Create sample sheet for a case."""
+        sample_sheet_content: list[list[Any]] = self.get_sample_sheet_content(case_id=case_id)
+        if not dry_run:
+            self.write_sample_sheet(
+                content=sample_sheet_content,
+                file_path=self.get_sample_sheet_path(case_id=case_id),
+                header=self.sample_sheet_headers,
+            )
+
+    def create_params_file(
+        self,
+        case_id: str,
+        dry_run: bool,
+    ):
+        """Create parameters file for a case."""
+        LOG.debug("Getting parameters information")
+        workflow_parameters: dict | None = self.get_workflow_parameters(case_id=case_id).dict()
+        if self._append_params_to_nextflow_config:
+            workflow_parameters = None
+        if not dry_run:
+            self.write_params_file(case_id=case_id, workflow_parameters=workflow_parameters)
```

Comment:
is the logic correct here?

```suggestion
        LOG.debug("Getting parameters information")
        workflow_parameters = None
        if self._append_params_to_nextflow_config:
            workflow_parameters: dict | None = self.get_workflow_parameters(case_id=case_id).dict()
        if not dry_run:
            self.write_params_file(case_id=case_id, workflow_parameters=workflow_parameters)
```

Reply:
is the opposite. Basically if the parameters are supposed to be written in the `nextflow config` then we don't want to write them in the `params file` so we set them to None to write an empty `params file`. I will leave this for now, but we can revisit this in the future. This will anyways change once all workflows are standarised.

Reply:
I see, but can we optimise it a bit? Typically, you initialise it to `None`, and then modify its value based on conditions. I think this approach would make it clearer and avoid confusion for someone who isn't familiar with the logic (like me 😆)
```suggestion
        LOG.debug("Getting parameters information")
        workflow_parameters = None
        if not self._append_params_to_nextflow_config:
            workflow_parameters: dict | None = self.get_workflow_parameters(case_id=case_id).dict()
        if not dry_run:
            self.write_params_file(case_id=case_id, workflow_parameters=workflow_parameters)
```

Reply:
makes sense. added